scripts/build-unix.sh

```diff
@@ -57,7 +61,7 @@ mkdir $BUILD_DIR
 cd $BUILD_DIR
 
 echo "Configuring..."
-cmake $SOURCE_DIR "$@"
+cmake $SOURCE_DIR ${BUILD_TYPE:+-DCMAKE_BUILD_TYPE=${BUILD_TYPE}} "$@"
```

Comment:
is this correct? 
or
```suggestion
cmake $SOURCE_DIR -DCMAKE_BUILD_TYPE=${BUILD_TYPE} "$@"
```

Reply:
It's to allow `BUILD_TYPE` to be empty without blowing up the script.

`${BUILD_TYPE:+-DCMAKE_BUILD_TYPE=${BUILD_TYPE}}` translates to:
If `BUILD_TYPE` is defined as a variable, expand it into `+-DCMAKE_BUILD_TYPE=${BUILD_TYPE}`
Otherwise, do nothing.

I figured `-DCMAKE_BUILD_TYPE=` would break stuff if empty.

Reply:
> but we want it to break if missing, right?

I thought one of the scripts deliberately omitted it, but apparently not - if it's always there your way is simpler.